### PR TITLE
feat(tui): delete confirmation overlay dialog

### LINKED ISSUE #46: vs-13c: TUI Confirm dialog — delete confirmation overlay
## Parent PRD

#1

## What to build

TUI confirmation dialog for destructive operations:

- Triggered by pressing `D` on a worktree in list view
- Centered modal showing: worktree name, path, warning about hooks, branch info
- `Enter` or `y` confirms deletion, `Esc` or `n` cancels
- After confirmation, executes remove and refreshes list
- Shows brief success/failure message

See FR-49.

### Layers touched
tui, tests

## Acceptance criteria

- [ ] `D` on list view opens confirm dialog for selected worktree
- [ ] Dialog shows worktree name, path, and warning
- [ ] Enter/y confirms, Esc/n cancels
- [ ] Removal executes on confirm
- [ ] List refreshes after removal
- [ ] Dialog is a centered overlay (not a full screen)

## Blocked by

- #33
- #17

## User stories addressed

- US-11 (TUI delete workflow)
- US-8 (safe removal with confirmation)

### PR DESCRIPTION
**Pull request by sadiksaifi**
Closes #46

## Summary
Implements a TUI confirmation dialog for destructive delete operations. Pressing `D` on the worktree list opens a centered modal overlay showing the worktree name, path, branch, and a hook warning. `Enter`/`y` confirms deletion (executing the remove command), `Esc`/`n` cancels. After deletion, a result screen shows success or failure before dismissing back to the refreshed list.

## User Stories Addressed
- US-11: TUI delete workflow — `D` triggers delete with confirmation dialog
- US-8: Safe removal with confirmation — dialog prevents accidental deletion

## Automated Testing
- Total tests added: 22
- Tests passing: 22
- Tests failing: 0
- `cargo clippy`: pass
- `cargo test`: pass (571 total: 563 unit + 8 integration)

## UAT Checklist
- [ ] Open TUI with `trench`, select a worktree, press `D` → confirm dialog appears as centered overlay
- [ ] Dialog shows the correct worktree name, path, and branch
- [ ] Dialog shows warning about pre-remove hooks
- [ ] Press `Esc` → dialog closes, no deletion occurs
- [ ] Press `n` → dialog closes, no deletion occurs
- [ ] Press `Enter` → worktree is deleted, result message shown
- [ ] Press `y` → worktree is deleted, result message shown
- [ ] After result message, press `Enter` → returns to list, deleted worktree is gone
- [ ] Press `D` on an empty list → nothing happens
- [ ] Delete a worktree with pre_remove hooks configured → hooks execute before deletion

## Known Issues
None

## Summary by CodeRabbit

* **New Features**
  * Added a delete-confirmation overlay for removing worktrees; invoke with D on a selected row, confirm with Enter or Y, cancel with N, Esc, or q.
  * Shows a post-delete result overlay with clear success/failure messages; Enter or Space dismiss results.
  * Global input clears overlays (Esc/q) before leaving the list.

* **Tests**
  * Added comprehensive tests covering the delete-confirm flow, result handling, key behavior, and edge cases.

### CODE REVIEW

**Comment by sadiksaifi** [comment]
@claude review

**Comment by sadiksaifi** [comment]
@claude Thanks for the thorough review! Here's the resolution for each item:

**Bug: Esc/q leave stale state** — Fixed in `1cc60fa`. Clear `delete_confirm_state` (and `sync_picker_state`) in the global Esc/q handler by checking the active screen before popping. Also handles Esc in result mode.

**Space undocumented in footer** — Fixed in `7876e24`. Updated `RESULT_FOOTER` to `" Enter/Space dismiss "`. Also applied the same fix to sync_picker's result footer for consistency.

**Long paths overflow dialog** — Fixed in `5fafed0`. Paths longer than the dialog's inner width are now truncated with a leading ellipsis.

**`screen_enum_has_six_variants` is fragile** — This test is intentionally a compile-time exhaustiveness guard. When a new `Screen` variant is added, it forces the developer to update this array, serving as a reminder to handle the new screen in key handling, rendering, etc. It's cheap to maintain and catches missed variants early. Keeping as-is.

**`buffer_text` loses row boundaries** — Acknowledged. The test helper concatenates cells without row separators, so theoretically a match could span rows. In practice, the dialog is centered in an 80+ column buffer with padding, making cross-row false positives extremely unlikely for the strings we assert on.

File: src/tui/screens/delete_confirm.rs [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+    // Path (truncate with leading ellipsis if too long for 60-col dialog)
+    let max_path_len = 56;
+    let path_display = if state.worktree_path.len() > max_path_len {
+        format!(
+            "\u{2026}{}",
+            &state.worktree_path[state.worktree_path.len() - (max_path_len - 1)..]
+        )
+    } else {
+        state.worktree_path.clone()
+    };
```
[the remark this answers is not in the document]
Comment (sadiksaifi) [reply src/tui/screens/delete_confirm.rs:96]: @coderabbitai

Fixed in `ca6ccef`.

Replaced byte-based slicing with character-aware truncation using `.chars().count()` and `.chars().skip()`. Non-ASCII worktree paths no longer risk a panic.

**Comment by sadiksaifi** [comment]
@coderabbitai

Fixed in `2d1e0b3`.

Changed the result dialog message area from `Constraint::Length(1)` to `Constraint::Min(1)` so multi-line error messages (e.g. from anyhow's `{e:#}` formatting) are fully visible instead of being truncated to the first line.

**Comment by sadiksaifi** [comment]
@coderabbitai

Fixed in `30dd79a`.

Added truncation for the name+branch line in the delete confirm dialog. Long branch names are now truncated with a trailing ellipsis to fit within the dialog width.
